src/Compilers/CSharp/Portable/Symbols/MethodSymbolExtensions.cs

```diff
@@ -222,8 +217,18 @@ internal static bool IsValidUnscopedRefAttributeTarget(this MethodSymbol method)
         {
             return !method.IsStatic &&
                 method.ContainingType?.IsStructType() == true &&
-                !method.IsConstructor() &&
+                method.MethodKind is (MethodKind.Ordinary or MethodKind.ExplicitInterfaceImplementation or MethodKind.PropertyGet or MethodKind.PropertySet) &&
```

Comment:
>method.MethodKind is (MethodKind.Ordinary or MethodKind.ExplicitInterfaceImplementation or MethodKind.PropertyGet or MethodKind.PropertySet) && [](http://example.com/codeflow?start=16&length=143)

What test scenario needed this change?

Reply:
Annotated local functions and lambdas within a struct - see `UnscopedRefAttribute_Method_05`.